examples/bri-3/src/bri/workgroup/worksteps/api/worksteps.controller.spec.ts

```diff
@@ -0,0 +1,193 @@
+      expect(createdWorkstep.version).toEqual(requestDto.version);
+      expect(createdWorkstep.status).toEqual(requestDto.status);
+      expect(createdWorkstep.workgroupId).toEqual(requestDto.workgroupId);
+      expect(createdWorkstep.securityPolicy).toEqual(requestDto.securityPolicy);
+      expect(createdWorkstep.privacyPolicy).toEqual(requestDto.privacyPolicy);
+    });
+  });
+
+  describe('getAllWorksteps', () => {
+    it('should return empty array if no worksteps', async () => {
+      // Act
+      const worksteps = await wController.getAllWorksteps();
+
+      // Assert
+      expect(worksteps.length).toEqual(0);
+    });
+
+    it('should return 2 worksteps if 2 exist', async () => {
+      // Arrange
+      const requestDto1 = { name: 'name1', version: 'version1', status: 'status1', workgroupId: 'wgid1', securityPolicy: 'secPolicy', privacyPolicy: 'privPolicy' } as CreateWorkstepDto;
+      const newWorkstepId1 = await wController.createWorkstep(requestDto1);
+
+      const requestDto2 = { name: 'name2', version: 'version2', status: 'status2', workgroupId: 'wgid2', securityPolicy: 'secPolicy', privacyPolicy: 'privPolicy' } as CreateWorkstepDto;
+      const newWorkstepId2 = await wController.createWorkstep(requestDto2);
+
+      // Act
+      const worksteps = await wController.getAllWorksteps();
+
+      // Assert
+      expect(worksteps.length).toEqual(2);
+      expect(worksteps[0].id).toEqual(newWorkstepId1);
+      expect(worksteps[0].name).toEqual(requestDto1.name);
+      expect(worksteps[0].version).toEqual(requestDto1.version);
+      expect(worksteps[0].status).toEqual(requestDto1.status);
+      expect(worksteps[0].workgroupId).toEqual(requestDto1.workgroupId);
+      expect(worksteps[0].securityPolicy).toEqual(requestDto1.securityPolicy);
+      expect(worksteps[0].privacyPolicy).toEqual(requestDto1.privacyPolicy);
+      expect(worksteps[1].id).toEqual(newWorkstepId2);
+      expect(worksteps[1].name).toEqual(requestDto2.name);
+      expect(worksteps[1].version).toEqual(requestDto2.version);
+      expect(worksteps[1].status).toEqual(requestDto2.status);
+      expect(worksteps[1].workgroupId).toEqual(requestDto2.workgroupId);
+      expect(worksteps[1].securityPolicy).toEqual(requestDto1.securityPolicy);
+      expect(worksteps[1].privacyPolicy).toEqual(requestDto1.privacyPolicy);
+    });
+  });
+
+  describe('createWorkstep', () => {
+    it('should throw BadRequest if name not provided', () => {
+      // Arrange
+      const requestDto = { version: 'version', status: 'status', workgroupId: 'wgid', securityPolicy: 'secPolicy', privacyPolicy: 'privPolicy' } as CreateWorkstepDto;
+
+      // Act and assert
+      expect(async () => {
+        await wController.createWorkstep(requestDto);
+      }).rejects.toThrow(new BadRequestException(NAME_EMPTY_ERR_MESSAGE));
+    });
+
+    it('should return new uuid from the created workstep when all params provided', async () => {
```

Comment:
test is a bit missleading in combination with test above, because it seems that only name is required
also for some optional fields workstep should be created too, so `when all params provided` is not correct

Reply:
this will be tackled as part of my issue on input DTO validation, no need to go in details here.

Reply:
Moved